src/main/java/frc/robot/subsystems/leds/LEDSubsystem.java

```diff
@@ -38,20 +49,80 @@ public void periodic() {
         // Toggle the blink boolean every duration to swap the LEDs between the driver piece color
         // and the blink color.
         if (blinkTimer.advanceIfElapsed(BLINK_DURATION_SECONDS)) blinking = !blinking;
-        ledStrip.setSolidColor(blinking ? BLINK_COLOR : color);
+        if(manual){
+            //if manual and continuous set the bottom to color
+            ledStrip.updateContinuousColor(color);
+            ledStrip.setContinuousColor();
+        } else {
+            setColorPulse();
+            //setTwoColor()
+        }
+    }
+
+    public void setManual(boolean manual){
+        this.manual = manual;
+    }
+
+    public void tagDetected(){
+        if(!aprilStarted){
+            aprilStarted = true;
+            aprilTimer.start();
+        }
+        if(aprilTimer.hasElapsed(APRIL_BLINK_DURATION * 3)){
+            aprilTimer.reset();
+            aprilTimer.start();
+        }
+        aprilTimer2.reset();
+        aprilTimer2.start();
+    }
+
+    public void setColorPulse(){
+        //current color is the color that will be added at the bottom of the strip buffer
+        if(!aprilTimer.hasElapsed(APRIL_BLINK_DURATION) && aprilTimer.hasElapsed(.001)){
+            ledStrip.updateContinuousColor(APRIL_COLOR);
+        } else {
+            ledStrip.fillContinuousColorIgnoringOneColor(color, APRIL_COLOR);
+            ledStrip.updateContinuousColor(color);
+        }
+        if(blinking){
+            ledStrip.fillContinuousColorIgnoringOneColor(BLINK_COLOR, APRIL_COLOR);
+        }
+        ledStrip.setContinuousColor();
```

Comment:
this is the same as line 53 in LEDStrip, and it feels weird to me when the same logic is called by different parts of the code. perhaps LEDSubsystem can just always call this at the end of periodic? so everything before that is just responsible for modifying the buffer, and then it's sent to the LEDs at the very end of periodic